src/algorithms/base_clustering.py

```diff
@@ -0,0 +1,49 @@
+import numpy as np
+
+
+class BaseClustering:
```

Comment:
this should be abstract class/interface without `__init__` method, use ABC or Protocol